tests/h/views/api/users_test.py

```diff
@@ -90,6 +88,11 @@ def test_raises_for_invalid_json_body(self, pyramid_request, patch):
         with pytest.raises(PayloadError):
             create(pyramid_request)
 
+    @pytest.fixture
+    def client_authority(self, patch):
+        patched = patch('h.views.api.users.client_authority')
+        patched.return_value = 'weylandindustries.com'
```

Comment:
This should also `return patched` in case any tests want to take the mock `client_authority` as an argument and use it (e.g. to access `client_authority.return_value`). (Even though no tests do actually do that yet, our patch fixtures always return the thing)

The usual style in our code is to name the local variable the same as the fixture too, so:

```python
@pytest.fixture
def client_authority(self, patch):
    client_authority = patch("h.views.api.users.client_authority")
    client_authority.return_value = "weylandindustries.com"
    return client_authority
```

Reply:
> As soon as I add a valid username, a valid email address, and an authority I start getting 200s back. But I get a 200 OK response and a user created whether the authority matches the auth client's authority or not. The user is created with the auth client's authority, and the different authority string passed in the request data is ignored. On master this would result in a {"status": "failure", "reason": "\'authority\' does not match authenticated client"} but on this branch it's ok: ❌

I agree I should reconcile this and make it match previous behavior (thanks for the catch). As an aside, I believe that the underlying problem is that the API takes an `authority` param at all—it doesn't really make sense if you think about it, right? There is no case in which it would be respected if it weren't the request's active authority (and it's unnecessarily redundant when it _does_ match). But we've got a published API, so, I'll make it behave like it used to.

Reply:
> As an aside, I believe that the underlying problem is that the API takes an authority param at all—it doesn't really make sense if you think about it, right?

I agree - I have no idea what that param is doing there! Unless we had envisioned possible multi- or cross-authority auth clients in the future. But afaik we have no such plans